The originally posted whine.el obviously contained literal control characters;
I was unable to find an undamaged source file. This patch tries to restore the
broken key sequences, hopefully it is close to the original. - ulm

--- whine.el~	2007-09-16 02:30:41.000000000 +0100
+++ whine.el	2007-11-19 10:03:47.000000000 +0100
@@ -237,5 +237,5 @@
                  "'"
                  )))
-  (whine "" (append whine-pain
+  (whine "\C-p" (append whine-pain
                '("Up, up, and away!"
                 "This sure is an exciting line!"
@@ -243,5 +243,5 @@
                 "Beginning of buffer"
                 )))
-  (whine "" (append whine-pain
+  (whine "\C-n" (append whine-pain
                '("Don't make me go that way"
                 "I don't want to go there!"
@@ -251,14 +251,14 @@
                 "File not found"
                 )))
-  (whine "" (append whine-pain
+  (whine "\C-d" (append whine-pain
                       '("Now what was that character again?"
                         "NOW look what you've done!"
                         "Buffer is read-only")))
-  (whine "" '("Oh, no!  I've forgotten what I did!"))
-  (whine "u" '("Oh, no!  I've forgotten what I did!"))
-  (whine "" '("Please don't make me KILL again!"))
-  (whine "" (append whine-pain
+  (whine "\C-_" '("Oh, no!  I've forgotten what I did!"))
+  (whine "\C-xu" '("Oh, no!  I've forgotten what I did!"))
+  (whine "\C-k" '("Please don't make me KILL again!"))
+  (whine "\C-f" (append whine-pain
                       '("Forward ho!" "Oh, no!")))
-  (whine "" (append whine-pain
+  (whine "\C-b" (append whine-pain
                 '("Please, Mr. Human Person, don't make me go that way!"
                 "I am taken aback!"
@@ -267,8 +267,8 @@
                 "Career aborted -- unable to restart."
                 )))
-  (whine "" (append whine-pain '("kill! Kill! KILL!")))
-  (whine "b" (append whine-pain '("This buffer is horrible!"
+  (whine "\C-w" (append whine-pain '("kill! Kill! KILL!")))
+  (whine "\C-xb" (append whine-pain '("This buffer is horrible!"
                  "I don't want to be here!")))
-  (whine "" '("You redrew the screen, you goon!"
+  (whine "\C-l" '("You redrew the screen, you goon!"
                 "No, you can't have a clear message line."
                 "This is getting boring."
@@ -277,17 +277,17 @@
                 ))
 
-  (whine "%" '("Oh wow!  I couldn't have done it better myself!"))
+  (whine "\M-%" '("Oh wow!  I couldn't have done it better myself!"))
   (whine "\C-i" '("Let's get the hell out of here!"))
   (whine "\e>" (append whine-pain '("It's off to the ends of the earth!")))
   (whine "\e<" (append whine-pain
                        '("It's off to the beginnings of the earth!")))
-  (whine "$" insult-word)
+  (whine "\M-$" insult-word)
   (whine " " insult-word)
-  (whine "" whine-insults)
-  (whine "b" whine-insults)
-  (whine "f" whine-insults)
-  (whine "" whine-insults)
-  (whine "2" (append whine-pain '("That just cuts me up.")))
-  (whine "x" '("You shouldn't have typed that command!"
+  (whine "\C-a" whine-insults)
+  (whine "\M-b" whine-insults)
+  (whine "\M-f" whine-insults)
+  (whine "\C-e" whine-insults)
+  (whine "\C-x2" (append whine-pain '("That just cuts me up.")))
+  (whine "\M-x" '("You shouldn't have typed that command!"
                  "That's not a legal command."
                  ))
@@ -300,5 +300,5 @@
                       "File accidentally deleted -- please retype."
                       ))
-  (whine "" (append whine-pain '("If you touch that key again I shall explode!"
+  (whine "\C-m" (append whine-pain '("If you touch that key again I shall explode!"
                 "If you touch that key again I shall freak out!"
                 "That's the space bar, not the return key!"
@@ -307,6 +307,6 @@
                 "You shouldn't have done that..."
                 "Now look what you've done!")))
-  (whine "" whine-pain)
-  (whine "v" whine-pain)
+  (whine "\C-v" whine-pain)
+  (whine "\M-v" whine-pain)
   ))
                 
